diff -Naur a/etc.Linux/os-release b/etc.Linux/os-release
--- a/etc.Linux/os-release	2016-09-02 01:55:25.000000000 +0500
+++ b/etc.Linux/os-release	2016-06-08 13:15:34.000000000 +0500
@@ -1,7 +1,7 @@
-NAME=Gentoo
-ID=gentoo
-PRETTY_NAME="Gentoo/Linux"
+NAME=Bentoo
+ID=bentoo
+PRETTY_NAME="Bentoo/Linux"
 ANSI_COLOR="1;32"
-HOME_URL="https://www.gentoo.org/"
-SUPPORT_URL="https://www.gentoo.org/support/"
-BUG_REPORT_URL="https://bugs.gentoo.org/"
+HOME_URL="https://bitbucket.org/redeyeteam/"
+SUPPORT_URL="https://bitbucket.org/redeyeteam/bentoo/wiki/"
+BUG_REPORT_URL="https://bitbucket.org/redeyeteam/bentoo/issues?status=new&status=open/"
